core/src/consensus/tendermint/engine.rs

```diff
@@ -238,6 +238,14 @@ impl ConsensusEngine for Tendermint {
 
         stake::on_term_close(block.state_mut(), block_number, &inactive_validators)?;
 
+        match term {
+            0 => {}
+            _ => match term_common_params.expect("Term common params should exist").era() {
+                0 => {}
+                1 => block.state_mut().snapshot_term_params()?,
+                _ => unimplemented!("It is not decided how we handle this"),
+            },
+        }
```

Comment:
0th term always have era=0
You can simplify this part as follows:
```
match term_common_params.expect("Term common params should exist").era() {
    0 => {}
    1 => block.state_mut().snapshot_term_params()?,
    _ => unimplemented!("It is not decided how we handle this"),
}
```

Reply:
`term_common_params` is still `None` when it closes first term.
I can try `term_common_params.map(CommonParams::era).unwrap_or(0)`

Reply:
Doesn't `clint.term_common_params` return `Some` except for the genesis block?

Reply:
No. it returns None when the term is 0. Leave this as is for now.